las/client.py

```diff
@@ -89,303 +90,871 @@ def _create_signing_headers(self, path: str):
+
+        >>> from las.client import BaseClient
+        >>> client = BaseClient()
+        >>> data = {'username': '<username>', 'password': '<password>'}
+        >>> client.update_secret('<secret id>', data, '<description>')
+
+        :param secret_id: Id of the secret
+        :type secret_id: str
+        :param data: Dict containing the data you want to keep secret
+        :type data: str
+        :param description: Description of the secret
+        :type description: Optional[str]
+        :return: Secret response from REST API
+        :rtype: dict
 
-        return self._make_request(requests.post, f'/documents/{document_id}', body={'feedback': feedback})
+        :raises: :py:class:`~las.InvalidCredentialsException`, :py:class:`~las.TooManyRequestsException`,\
+ :py:class:`~las.LimitExceededException`, :py:class:`requests.exception.RequestException`
+        """
+        body = {
+            'data': data,
+            'description': description,
+        }
+        return self._make_request(requests.patch, f'/secrets/{secret_id}', body=dictstrip(body))
 
-    def delete_consent(self, consent_id: str) -> dict:
-        """Delete documents with this consent_id, calls the DELETE /consents/{consentId} endpoint.
+    def create_transition(self, name, transition_type: str, in_schema: dict, out_schema: dict,
+                          params: Optional[dict] = None, description: Optional[str] = None) -> dict:
+        """Creates a transition handle, calls the POST /transitions endpoint.
 
+        >>> import json
+        >>> from pathlib import Path
         >>> from las.client import BaseClient
         >>> client = BaseClient()
-        >>> client.delete_consent('<consent id>')
+        >>> in_schema = {'$schema': 'https://json-schema.org/draft-04/schema#', 'title': 'in', 'properties': {...} }
+        >>> out_schema = {'$schema': 'https://json-schema.org/draft-04/schema#', 'title': 'out', 'properties': {...} }
+        >>> # A typical docker transitions
+        >>> docker_params = {'imageUrl': '<image_url>', 'credentials': {'username': '<username>', 'password': '<password>'}}
+        >>> client.create_transition('<name>', 'docker', in_schema, out_schema, docker_params)
+        >>> # A typical manual transitions
+        >>> assets = {'jsRemoteComponent': 'las:asset:<hex-uuid>', '<other asset name>': 'las:asset:<hex-uuid>'}
+        >>> manual_params = {'assets': assets}
+        >>> client.create_transition('<name>', 'manual', in_schema, out_schema, manual_params)
+
+        :param in_schema: Json-schema that defines the input to the transition
+        :type in_schema: dict
+        :param out_schema: Json-schema that defines the output of the transition
+        :type out_schema: dict
+        :param name: Name of the transition
+        :type name: str
+        :param transition_type: Type of transition "docker"|"manual"
+        :type transition_type: str
+        :param params: Extra parameters to the transition
```

Comment:
Does 'Extra' sounds too generic? "Parameters to corresponding transition_type." + An overview of the supported parameters by transition_type?